### PR TITLE
[Merged by Bors] - Flag disable post verification

### PR DESCRIPTION
**Pull request by poszu**
## Motivation
There are setups with one _public_ node that "talks to the Internet" and stands in front of more _private_ nodes that only connect to that _public_ node. In such a setup, the private nodes can trust that the public one already verified all data coming from the network and it is not required to repeat that work. The PoST proofs of ATXs are especially heavy.

## Description
- added a flag `smeshing-opts-verifying-disable` and associated config field,
- created a noop post verifier if the flag is set,
- additionally, removed unused field from activation Handler.

## TODO

- [x] Explain motivation or link existing issue(s)
- [x] Test changes and document test plan
- [x] Update documentation as needed
- [x] Update [changelog](../CHANGELOG.md) as needed

### CODE REVIEW

**Comment by ivan4th** [comment]
Wouldn't it make sense to disable fetch server for the unverified data on the nodes with this option, either always or at least by default, to lessen the effect of possible "spreading" of unverified POST proofs across "careless" nodes that enable this option? Granted, it will be limited by other "proper" nodes of course, but still I'd be a bit proactive in eliminating such possible side effect. Or maybe require private network option together with this one

File: CHANGELOG.md [hunk 1]
```diff
@@ -89,6 +89,9 @@ configuration is as follows:
 ### Features
 
 * [#5462](https://github.com/spacemeshos/go-spacemesh/pull/5462) Add separate metric for failed p2p server requests
+* [#5517](https://github.com/spacemeshos/go-spacemesh/pull/5517)
+  Added a flag `--smeshing-opts-verifying-disable` meant for disabling verifying POST in incoming ATXs on
```
Comment (fasmat) [inline CHANGELOG.md]: Should we also mention that this flag can be set via config?

**Comment by poszu** [comment]
> Wouldn't it make sense to disable fetch server for the unverified data on the nodes with this option, either always or at least by default, to lessen the effect of possible "spreading" of unverified POST proofs across "careless" nodes that enable this option? Granted, it will be limited by other "proper" nodes of course, but still I'd be a bit proactive in eliminating such possible side effect. Or maybe require private network option together with this one

It makes sense to allow this flag together with `--p2p-private-network` only.

**Review by fasmat — APPROVED** [review]

**Comment by fasmat** [comment]
> Wouldn't it make sense to disable fetch server for the unverified data on the nodes with this option, either always or at least by default, to lessen the effect of possible "spreading" of unverified POST proofs across "careless" nodes that enable this option? Granted, it will be limited by other "proper" nodes of course, but still I'd be a bit proactive in eliminating such possible side effect. Or maybe require private network option together with this one

To be fair if a node has verification disabled and broadcast an unverified ATX with invalid proof the situation is no different from a node receiving this invalid proof from "the source". Nodes should not blindly trust incoming data, only other nodes connected to the public network with verification disabled would be affected by this.

In terms of generating additional network traffic with invalid ATXs: this is no different from a node just broadcasting random ATXs, or am I mistaken?

**Comment by poszu** [comment]
> > Wouldn't it make sense to disable fetch server for the unverified data on the nodes with this option, either always or at least by default, to lessen the effect of possible "spreading" of unverified POST proofs across "careless" nodes that enable this option? Granted, it will be limited by other "proper" nodes of course, but still I'd be a bit proactive in eliminating such possible side effect. Or maybe require private network option together with this one
> 
> To be fair if a node has verification disabled and broadcast an unverified ATX with invalid proof the situation is no different from a node receiving this invalid proof from "the source". Nodes should not blindly trust incoming data, only other nodes connected to the public network with verification disabled would be affected by this.
> 
> In terms of generating additional network traffic with invalid ATXs: this is no different from a node just broadcasting random ATXs, or am I mistaken?

@mafa Well yes. @ivan4th proposed to add a check to avoid misusing this option by unaware users. It's intended for use in private setups only so it makes sense to reject it if `--p2p-private-network` is not set.

**Comment by poszu** [comment]
@ivan4th updated, PTAL

**Review by ivan4th — APPROVED** [review]
LGTM

**Comment by poszu** [comment]
bors merge

File: CHANGELOG.md [hunk 2]
```diff
@@ -120,6 +120,11 @@ configuration is as follows:
   Reduce the default number of CPU cores that are used for verifying incoming ATXs to half of the available cores.
 
 * [#5462](https://github.com/spacemeshos/go-spacemesh/pull/5462) Add separate metric for failed p2p server requests
+* [#5517](https://github.com/spacemeshos/go-spacemesh/pull/5517)
```
Comment (fasmat) [inline CHANGELOG.md]: That change will be in v1.3.8 not v1.3.6 😉

**Comment by poszu** [comment]
bors cancel

File: CHANGELOG.md (hunk 2 above)
Comment (poszu) [reply CHANGELOG.md]: Fixed, thanks!

**Comment by poszu** [comment]
bors merge

**Comment by poszu** [comment]
bors cancel

**Comment by pigmej** [comment]
Guys, but that flag `--p2p-private-network` has nothing to do with public/private nodes setup.

**Comment by poszu** [comment]
bors merge

**Comment by poszu** [comment]
bors cancel

**Comment by poszu** [comment]
bors merge

**Comment by poszu** [comment]
bors cancel

**Comment by poszu** [comment]
bors merge

**Pull request closed without merging**